ng-dev/pr/merge/fetch-pull-request.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {params, types as graphqlTypes} from 'typed-graphqlify';
+import {AuthenticatedGitClient} from '../../utils/git/authenticated-git-client';
+import {getPr} from '../../utils/github';
+
+/** Graphql schema for the response body the requested pull request. */
+const PR_SCHEMA = {
+  url: graphqlTypes.string,
+  isDraft: graphqlTypes.boolean,
+  state: graphqlTypes.oneOf(['OPEN', 'MERGED', 'CLOSED'] as const),
+  number: graphqlTypes.number,
+  // Only the last 100 commits from a pull request are obtained as we likely will never see a pull
+  // requests with more than 100 commits.
+  commits: params(
+    {last: 100},
+    {
+      totalCount: graphqlTypes.number,
+      nodes: [
+        {
+          commit: {
+            status: {
+              state: graphqlTypes.oneOf(['FAILURE', 'PENDING', 'SUCCESS'] as const),
+            },
+            message: graphqlTypes.string,
+          },
+        },
+      ],
+    },
+  ),
+  baseRefName: graphqlTypes.string,
+  title: graphqlTypes.string,
+  labels: params(
+    {first: 100},
+    {
+      nodes: [
+        {
+          name: graphqlTypes.string,
+        },
+      ],
+    },
+  ),
+};
+
+/** A pull request retrieved from github via the graphql API. */
+export type RawPullRequest = typeof PR_SCHEMA;
+
+/** Fetches a pull request from Github. Returns null if an error occurred. */
+export async function fetchPullRequestFromGithub(
```

Comment:
I know that this is just a refactoring of the function into a separate file, but looking at it now, I think we should have this file just export the schema for the GraphQL query and then we can just directly call `await getPr(PR_SCHEMA, prNumber, git)` in the `ng-dev/pr/merge/pull-request.ts` file.  `getPr` is already set up to use the type as the outputted type.

Reply:
Yeah, I think having the wrapper function allows us to nicely isolate the pull request fetching into this file, while otherwise we'd manually need to feed it the schema. I guess it doesn't matter too much.

Reply:
I'd like to keep it just for the sake of isolating everything into the `fetch-pull-request.ts` file here. we can revisit if you feel strongly.

Reply:
We can keep it there, it's something we could also look at during the larger refactor outlined in #203 to see if it makes sense when all of that settles.

Reply:
Agreed 👍